src/main_functions_save.c

```diff
@@ -4,68 +4,123 @@
 #include <assert.h>
 #include <shellstrings.h>
 #include <string.h>
+#include "parser.h"
 #include "main_functions_save.h"
+#include "log.c/src/log.h"
 
 /* See main_functions_save.h */
-void save_corrections(char* filename, char** lines)
-{
-	FILE* f = fopen(filename,"w");
+void save_corrections(char *filename, char **lines) {
+    log_trace("save_corrections opening save file destination '%s'.", filename);
+	FILE *f = fopen(filename, "w");
     assert(f != NULL);
+
 	int i = 0;
+
 	while (lines[i] != NULL) {
+        log_trace("Writing line into file '%s'.", lines[i]);
 		fprintf(f, "%s", lines[i]); // write lines into file
-		i++;
+		
+        i++;
 	}
+
+    log_trace("save_corrections closing save file destination '%s'.", filename);
 	fclose(f);
 }
 
 /* See main_functions_save.h */
-void save_page(char* filename, char** lines, bool* quit)
-{
+void save_page(char *filename, char **lines, bool *quit, bool* color) {
```

Comment:
Why is this a `bool *` and not just a `bool`?

Reply:
`*quit` and `*color` are booleans called here because the save_mode should change its value, initially malloc'd in main.c, in the case that user chooses to start editing again and the large while loop in main.c must begin again. If I just called the value, I wouldn't be able to change it to affect future prints to output. main_page is one of a few functions that can change that value.
I've changed the function calls for functions who won't affect the value, such as the many you commented on for shellstrings, but I think for batch_mnode it might be necessary to do `*quit` and `*color`.